model/jmb_enest_nlparam.jl

```diff
@@ -0,0 +1,807 @@
+@NLparameter(cge, klem_bar[r in regions, s in sectors] == value(vaen_bar[r, s]) + value(ne_bar[r, s]));
+
+@NLparameter(cge, alpha_l[r in regions, s in sectors] == value(ld0_p[r, s]) / value(va_bar[r, s]));
+@NLparameter(cge, alpha_k[r in regions, s in sectors] == value(kd0_p[r, s]) / value(va_bar[r, s]));
+@NLparameter(cge, theta_fe[r in regions, g in fe, s in sectors] == value(id0_p[r, g, s]) / value(fe_bar[r, s]));
+@NLparameter(cge, theta_en[r in regions, g in en, s in sectors] == value(id0_p[r, g, s]) / value(en_bar[r, s]));
+@NLparameter(cge, theta_ele[r in regions, s in sectors] == sum(value(id0_p[r, g, s]) for g in ele) / value(en_bar[r, s]));
+@NLparameter(cge, theta_ne[r in regions, g in ne, s in sectors] == value(id0_p[r, g, s]) / value(ne_bar[r, s]));
+
+@NLparameter(cge, theta_va[r in regions, s in sectors] == value(va_bar[r, s]) / (value(va_bar[r, s]) + value(en_bar[r, s])));
+@NLparameter(cge, theta_ene[r in regions, s in sectors] == value(en_bar[r, s]) / (value(va_bar[r, s]) + value(en_bar[r, s])));
+@NLparameter(cge, theta_kle[r in regions, s in sectors] == value(en_bar[r, s]) / (value(vaen_bar[r, s]) + value(klem_bar[r, s])));
+@NLparameter(cge, theta_mm[r in regions, s in sectors] == value(en_bar[r, s]) / (value(ne_bar[r, s]) + value(klem_bar[r, s])));
+
+@NLparameter(cge, esub_fe == 0.5);
+@NLparameter(cge, esub_ele == 0.1);
+@NLparameter(cge, esub_ve == 0.5);
+@NLparameter(cge, esub_ne == 0.2);
+@NLparameter(cge, esub_klem == 0.3);
+
+
+replace_nan_inf(alpha_kl)
+replace_nan_inf(alpha_x)
+replace_nan_inf(alpha_d)
+replace_nan_inf(alpha_n)
+replace_nan_inf(theta_n)
+replace_nan_inf(theta_m)
+
+replace_nan_inf(alpha_l)
+replace_nan_inf(alpha_k)
+replace_nan_inf(theta_fe)
+replace_nan_inf(theta_en)
+replace_nan_inf(theta_ele)
+replace_nan_inf(theta_ne)
+replace_nan_inf(theta_va)
+replace_nan_inf(theta_ene)
+replace_nan_inf(theta_kle)
+replace_nan_inf(theta_mm)
+
+################
+# VARIABLES
+################
+
+sv = 0.00
+sub_set_y = filter(x -> y_check[x] != 0.0, combvec(regions, sectors));
+sub_set_x = filter(x -> haskey(s0, x), combvec(regions, goods));
+sub_set_a = filter(x -> a_set[x[1], x[2]] != 0.0, combvec(regions, goods));
+sub_set_pa = 
+    filter(x -> haskey(a0, (x[1], x[2])), combvec(regions, goods));
+sub_set_pd =
+    filter(x -> haskey(xd0, (x[1], x[2])), combvec(regions, goods));
+sub_set_pk =
+    filter(x -> haskey(kd0, (x[1], x[2])), combvec(regions, sectors));
+sub_set_py = filter(x -> y_check[x[1], x[2]] >= 0, combvec(regions, goods));
+
+#energy subsetting
+en_bar0 = Dict();
+va_bar0 = Dict();
+[en_bar0[r,s] = value(en_bar[r,s]) for r in regions for s in sectors];
+[va_bar0[r,s] = value(va_bar[r,s]) for r in regions for s in sectors];
+    
+sub_set_pe = filter(x -> en_bar0[x[1], x[2]] != 0.0, combvec(regions, sectors));
+sub_set_pva = filter(x -> va_bar0[x[1], x[2]] != 0.0, combvec(regions, sectors));
+sub_set_va = sub_set_pva;
+sub_set_e = sub_set_pe;
```

Comment:
### Consistency, Concision
- [ ] It would be great to define these subsets when we're setting up for the model like we did for `_model_set!`. Try tackling this following that approach!